Add Prometheus service monitor for flagger-loadtester

branden opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://github.com/mesosphere/kubernetes-base-addons/blob/master/CONTRIBUTING.md
2. When you're changing an existing addon, please do so with at least 2 commits:

   1. create a copy of the addon spec file without doing any changes
   2. change the copy

   That way it’s much easier to review what actually has been changed.
-->

**What type of PR is this?**
<!-- Bug, Chore, Documentation, Feature -->
Bug

**What this PR does/ why we need it**:
<!-- Explain, without going into the details, what this PR does, and what problem it solves. -->
This adds a service monitor and related service labels so that Prometheus can discover flagger-loadtester's metrics endpoint.

I tested this PR by deploying KBA from this branch, browsing to `/ops/portal/prometheus/targets`, and observing that the flagger-loadtester target is present and available:

<img width="851" alt="Screen Shot 2020-10-30 at 12 10 54 PM" src="https://user-images.githubusercontent.com/42242/97749868-032edf80-1aad-11eb-87ba-21fcd22d6f90.png">

**Which issue(s) this PR fixes**:
<!-- Add a link to the JIRA issue. Otherwise, put "no issue."
* jql=key in (D2IQ-<NUMBER>)
* https://jira.d2iq.com/browse/D2iQ-<NUMBER>
-->
https://jira.d2iq.com/browse/D2IQ-72694

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
Restores metrics collection for flagger-loadtester.
```

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [x] The code builds and passes lint/style checks locally.
* [x] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [x] The documentation is updated where needed.

gracedo reviewed (approved):
Would you be able to verify this manually on a cluster that these metrics are getting scraped into prom?

branden commented:
Looks like https://github.com/mesosphere/charts/pull/860 broke the flagger chart when I changed the CRD resource version without updating the spec to match the new schema. I'll need to fix that before this PR can proceed.

armandgrillet commented on addons/flagger/flagger.yaml:
```diff
@@ -8,9 +8,9 @@ metadata:
   annotations:
     sidecar.istio.io/inject: "false"
     appversion.kubeaddons.mesosphere.io/flagger: "0.19.0"
-    catalog.kubeaddons.mesosphere.io/addon-revision: "0.19.0-1"
```
If we bump the chartReference.version to `0.20.1`, should we update this and the `appversion.kubeaddons.mesosphere.io/flagger` to `0.20` too?

branden replied:
I think it was a coincidence that the app and chart version were the same. Chart version 0.20.1 still deploys flagger 0.19.0: https://github.com/mesosphere/charts/blob/659d0ddada7a74e57e6a73f5a0de01fcf633b151/staging/flagger/Chart.yaml#L3-L4

s12chung replied:
I agree with @branden. `appversion` is the internal app version inside of the chart.

joejulian commented:
/test

branden commented:
The teamcity job is failing because of a bug in the chart that's already present: https://jira.d2iq.com/browse/D2IQ-73316 This bug occurs when flagger is deployed, removed, and deployed again. This PR is triggering it because it's the first change to flagger since we added a set of tests that causes flagger to be deployed a second time to the test cluster.
